### PR TITLE
MYDP-386 Add region info

### PR DESCRIPTION
**Pull request by erik-deveza**
This PR adds region info and NZ postcode searching

- Apply ObjectMother design pattern (via PostcodeMother)
- Added 2 entities for region info (NZ and AU)
- Improved test readability

Reference:
https://github.com/MyDesktop/myd-web-app/blob/master/cgi-bin/ajax/getsuburbs.cgi

### CODE REVIEW

File: src/main/java/au/com/mydesktop/postcode/data/entity/Postcode.kt [hunk 1]
```diff
@@ -1,47 +1,206 @@
...
-        @Column(name = "nzcoredistrict") var nzCoreDistrict: Int?):Serializable
+        @Column(name = "nzcoredistrict") var nzCoreDistrict: Int?) : Serializable {
+    class Builder {
+        private var overseasCountryRegion: OverseasCountryRegion? = null
+        private var nzCityId: Int? = null
+        private var nzCoreDistrict: Int? = null
+
+        fun overseasCountryRegion(overseasCountryRegion: OverseasCountryRegion?) = apply { this.overseasCountryRegion = overseasCountryRegion }
+        fun nzCityId(nzCityId: Int?) = apply { this.nzCityId = nzCityId }
+        fun nzCoreDistrict(nzCoreDistrict: Int?) = apply { this.nzCoreDistrict = nzCoreDistrict }
```
Comment (MingjianSun) [inline src/main/java/au/com/mydesktop/postcode/data/entity/Postcode.kt]: I am just curious what is the return type, compared to java, it should be `this`?

**Comment by MingjianSun** [comment]
Could anyone re-configure this repo which does not allowed merge pr before approval?

File: src/main/java/au/com/mydesktop/postcode/data/entity/Postcode.kt (hunk 1 above)
Comment (erik-deveza) [reply src/main/java/au/com/mydesktop/postcode/data/entity/Postcode.kt]: This is the implementation of apply
image

**Comment by erik-deveza** [comment]
Thanks @MingjianSun !

File: src/main/java/au/com/mydesktop/postcode/data/entity/Postcode.kt (hunk 1 above)
Comment (MingjianSun) [reply src/main/java/au/com/mydesktop/postcode/data/entity/Postcode.kt]: That is beautiful, I strongly want to upgrade our sonarqube version to support kotlin.

**Pull request merged**